<launch>
<!-- Arguments -->
  <arg name="odom_frame_id"  default="odom"/>
  <arg name="base_frame_id"  default="base_link"/>
  <arg name="scan_topic"     default="scan"/>
  <arg name="initial_pose_x" default="3.0"/>
  <arg name="initial_pose_y" default="-4.0"/>
  <arg name="initial_pose_a" default="-2.187"/>
  <arg name="map_file" default="$(find sicknav350)/maps/houdar_AI_home_20180801.yaml"/>
 <!-- launch sick  -->
 <include file="$(find sicknav350)/launch/sicknav350.launch" />
 <!--Map Server -->
 <node name="map_server" type="map_server" pkg="map_server" args="$(arg map_file)" />

  <arg name="one_deg" value="0.017453" />
  <!-- AMCL -->
  <node pkg="amcl" type="amcl" name="amcl" respawn="true" output="screen">
    <!--overal para -->
    <param name="min_particles"             value="500"/>
    <param name="max_particles"             value="3000"/>
    <param name="kld_err"                   value="0.02"/>
    <!-- dist change 5cm -->
    <param name="update_min_d"              value="0.05"/>
    <!-- angle change 6 deg(1 deg = 0.017 rad) -->
    <param name="update_min_a"              value="0.05"/>
    <param name="resample_interval"         value="1"/>
    <param name="transform_tolerance"       value="0.5"/>
    <param name="recovery_alpha_slow"       value="0.001"/>
    <param name="recovery_alpha_fast"       value="0.1"/>
    <param name="initial_pose_x"            value="$(arg initial_pose_x)"/>
    <param name="initial_pose_y"            value="$(arg initial_pose_y)"/>
    <param name="initial_pose_a"            value="$(arg initial_pose_a)"/>
    <param name="gui_publish_rate"          value="50.0"/>
    <!-- scan para -->
    <remap from="scan"                      to="$(arg scan_topic)"/>
    <param name="laser_min_range"           value="1"/>
    <param name="laser_max_range"           value="30"/>
     <!-- correspond to laser scan set para -->
    <param name="laser_max_beams"           value="1440"/>
    <!--  -->
    <param name="laser_z_hit"               value="0.5"/>
    <param name="laser_z_rand"              value="0.5"/>
    <param name="laser_z_short"             value="0.05"/>
    <param name="laser_z_max"               value="0.05"/>
    <param name="laser_sigma_hit"           value="0.2"/>
    <param name="laser_lambda_short"        value="0.1"/>
    <param name="laser_likelihood_max_dist" value="2.0"/>
    <param name="laser_model_type"          value="likelihood_field"/>
    <!-- odom para --> 
    <!--  !!wheel control type!!  -->
    <param name="odom_model_type"           value="diff"/>
    <param name="odom_alpha1"               value="0.1"/>
    <param name="odom_alpha2"               value="0.1"/>
    <param name="odom_alpha3"               value="0.1"/>
    <param name="odom_alpha4"               value="0.1"/>
    <param name="odom_frame_id"             value="$(arg odom_frame_id)"/>
    <param name="base_frame_id"             value="$(arg base_frame_id)"/>
  </node>
</launch>
